be/src/vec/functions/array/function_array_match_any.cpp

```diff
@@ -0,0 +1,292 @@
+                param_value->type, &param_value->value, query_param));
+        // here should use MATCH_ANY_QUERY
+        RETURN_IF_ERROR(iter->read_from_inverted_index(
+                data_type_with_name.first, query_param->get_value(),
+                segment_v2::InvertedIndexQueryType::MATCH_ANY_QUERY, num_rows, roaring));
+
+        segment_v2::InvertedIndexQueryCacheHandle null_bitmap_cache_handle;
+        RETURN_IF_ERROR(iter->read_null_bitmap(&null_bitmap_cache_handle));
+        std::shared_ptr<roaring::Roaring> null_bitmap = null_bitmap_cache_handle.get_bitmap();
+        if (null_bitmap) {
+            *bitmap -= *null_bitmap;
+        }
+        *bitmap = *roaring;
+        param_value->is_evaled_inverted_idx = true;
+        return Status::OK();
+    }
+
+    Status execute_impl(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
+                        size_t result, size_t input_rows_count) const override {
+        DBUG_EXECUTE_IF("match.invert_index_not_support_execute_match", {
+            return Status::Error<ErrorCode::INVERTED_INDEX_NOT_SUPPORTED>(
+                    "FunctionMatchAny not support execute_match");
+        })
+        bool is_eval_inverted_index = false;
+        auto* param_value = reinterpret_cast<ParamValue*>(
+                context->get_function_state(FunctionContext::FRAGMENT_LOCAL));
+        if (param_value != nullptr) {
+            is_eval_inverted_index = param_value->is_evaled_inverted_idx;
+        }
+
+        // the array column in block should prepared
+        auto left_column =
+                block.get_by_position(arguments[0]).column->convert_to_full_column_if_const();
+        const ColumnArray* array_column = nullptr;
+        const UInt8* array_null_map = nullptr;
+        if (left_column->is_nullable()) {
+            auto nullable_array = reinterpret_cast<const ColumnNullable*>(left_column.get());
+            array_column =
+                    reinterpret_cast<const ColumnArray*>(&nullable_array->get_nested_column());
+            array_null_map = nullable_array->get_null_map_column().get_data().data();
+        } else {
+            array_column = reinterpret_cast<const ColumnArray*>(left_column.get());
+        }
+        const auto& offsets = array_column->get_offsets();
+        // prepare dst column
+        auto dst = ColumnVector<UInt8>::create(offsets.size());
+        auto& dst_data = dst->get_data();
+        auto dst_null_column = ColumnUInt8::create(offsets.size());
+        auto& dst_null_data = dst_null_column->get_data();
+
+        if (is_eval_inverted_index) {
+            // here is no need to eval array column again,just put array column to res
+            for (size_t row = 0; row < offsets.size(); ++row) {
+                if (array_null_map && array_null_map[row]) {
+                    dst_null_data[row] = true;
+                    continue;
+                }
+                dst_null_data[row] = false;
```

Comment:
Put it in else branch and remove continue.

Reply:
done!